enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+
+Expected work:
+
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * set the appropriate Ironic-backed Nova flavor in `install-config.yaml`
+
+#### Case 3. Install with both VM and BM workers (BMs on a preexisting network)
+
+This case leverages [Provider Networks and Custom
+Subnets](openstack-customer-provided-subnets.md). The cluster is installed on
+the same pre-existing network where the BMs boot.
+
+Prerequisites for the user:
+
+* Being able to provision VMs nodes on the preexisting subnet where BMs run.
+
+Expected work:
+
+* Enable `install-config.yaml` to allow more than one worker set.
+* Adapt the wait-for install-complete timeout, either in code or in
+  documentation, in order to accomodate for the BM booting times.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * specify two different worker sets in `install-config.yaml`: one for the VM
+    MachineSet, and one with Ironic-backed Nova flavors for the BM MachineSet
+
+#### Case 4. Install on VMs then add BM workers (BMs on a preexisting network)
+
+In this case, the cluster is installed on the installer-provisioned network.
+The additional MachineSet bears the `subnetName` of the preexisting network
+onto which the BMs will boot.
+
+Prerequisites for the user:
+
+* Connectivity must be ensured between the preexisting subnet and the cluster's
+  subnet (eg through a router).
+
+Expected work:
+
+* Allow configuring a Neutron subnet for a given worker node pool.
+* Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+* Document how to use:
+  * make sure the `install-config.yaml` `MachineCIDR` contains the BMs network
+  * add a MachineSet with an Ironic-backed Nova flavor and the correct
+    `subnetName`.
+
+#### Adapt the wait-for install-complete timeout
+
+The last step of the installer is to check for the availability of the cluster
+and report a failed installation if a timeout is exceeded. However, BM nodes
+take longer than VMs to boot and may cause a timeout even in the case of a
+perfectly healthy cluster. With the proposed implementation of baremetal
+workers, the Installer is not aware whether the worker nodes are VMs or BMs; a
+conditional timeout extention is therefore not possible.
+
+The most straightforward solution would be to let the user set a custom timeout
+when installing, or disable the timeout altogether.
+
+#### Adapt the Certificate-Signing-Request timeouts to the BM booting times.
+
+In order to be authorised to join the cluster, new workers are expected to
+issue a CSR within a predefined timeout. This timeout currently might not
+account for BM boot times.
+
+#### Allow configuring a Neutron subnet for a given worker node pool
+
```

Comment:
Do we really want to do this? shouldn't security group creation be left to user as well?  Connectivity between subnets and security group configuration are tightly coupled.

Reply:
Added as a user requirement.